### DISCUSSION TITLE
Svg icon is not properly rendered in Typst output

### DISCUSSION THREAD

**Discussion by oliviermeslin** [quarto-dev/quarto-cli#10962] [Q&A] [answered]
### Description

I just discovered that the official SVG Python logo is not properly rendered in Typst output (the bottom and the right parts are missing), whereas it is fine in html output. I do not know whether this is related to the Quarto -> Typst conversion process, or to an internal Typst issue, or to the SVG file itself. Do you have an idea what might be going on?

Below are two MWE. 

```
---
title: "The SVG is fine in html output"
format: html
---

![](https://s3.dualstack.us-east-2.amazonaws.com/pythondotorg-assets/media/files/python-logo-only.svg)
```

```
---
title: "The SVG is cut in Typst output"
format: typst
---

![](https://s3.dualstack.us-east-2.amazonaws.com/pythondotorg-assets/media/files/python-logo-only.svg)
```

Results of `quarto check`:

```
Quarto 1.5.57
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.57
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.09
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/onyxia/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.5 (Conda)
      Path: /opt/conda/bin/python
      Jupyter: 5.7.2
      Kernels: python3

(/) Checking Jupyter engine render....Traceback (most recent call last):
  File "/opt/quarto/share/jupyter/jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "/opt/quarto/share/jupyter/notebook.py", line 18, in <module>
    import nbformat
ModuleNotFoundError: No module named 'nbformat'
[✓] Checking Jupyter engine render....OK
```

**Comment by cderv** [comment, marked as answer]
I believe this is related to typst handling of SVG. 

When providing a svg through a URL, quarto does download the SVG. The version download is correct. Here it is: 

![python-logo-only](https://github.com/user-attachments/assets/4f98b8d3-55bb-41cf-82bb-979de3c034f7)

But when loading this logo file in https://typst.app and using it in a document 
````typst
#image("python-logo-only.svg")
````

![image](https://github.com/user-attachments/assets/1a284301-b0d9-405e-a73b-45a09b6aaace)

Even playing with size or `fit` option I can't make this works. 

SVG is kind of new in typst. In their doc: https://typst.app/docs/reference/visualize/image/, they say
> Note: Work on SVG export is ongoing and there might be visual inaccuracies in the resulting PDF. Make sure to double-check embedded SVG images. If you have an issue, also feel free to report it on [GitHub](https://github.com/typst/typst/issues?q=is%3Aopen+is%3Aissue+label%3Asvg).

I don't see any issue on size. but maybe this python logo has some specific content that makes typst trip. 

This would probably be something to see with them first.

**Reply by oliviermeslin** [reply]
Thanks @cderv !

**Reply by cderv** [reply]
Will you open a feedback to Typst repo directly ?

**Reply by oliviermeslin** [reply]
Yes I will!

**Reply by oliviermeslin** [reply]
@cderv : I opened a Typst [issue](https://github.com/typst/typst/issues/5121)

**Reply by cderv** [reply]
Thanks !

**Reply by oliviermeslin** [reply]
@cderv : The problem is solved in the coming 0.12.0 version of Typst.

**Reply by mcanouil** [reply]
This will likely be for 1.7 then.
